spec/activerecord-bitemporal/bitemporal_scope_spec.rb

```diff
@@ -101,7 +101,13 @@ class Article < ActiveRecord::Base
       not_have_valid_at(table: table).and not_have_transaction_at(table: table)
     }
 
-    around { |e| Timecop.freeze(time_current) { e.run } }
+    around { |e|
+      if time_current
+        Timecop.freeze(time_current) { e.run }
+      else
+        e.run
```

Comment:
removed it because it did not bug reproduce in `Timecop.freeze`.

Reply:
Could you explain why this is? Perhaps I don't understand the condition where `Arel::Nodes::And#children` is more than 3 nodes.

Reply:
`.or` removes the same query.
So when you do `Timecop.freeze`, the time is fixed and the `.or` query is removed.

```ruby
Timecop.freeze(Time.current) {
  puts Company.or(Company.all).to_sql
}
__END__
output:
SELECT "companies".*
  FROM "companies"
 WHERE "companies"."transaction_from" <= '2022-08-24 08:15:41.962649'
   AND "companies"."transaction_to"   >  '2022-08-24 08:15:41.962649'
   AND "companies"."valid_from"       <= '2022-08-24 08:15:41.962649'
   AND "companies"."valid_to"         >  '2022-08-24 08:15:41.962649'
```

However, if you do not `Timecop.freeze`, the `.or` query will not be removed because the time is different.

```ruby
puts Company.or(Company.all).to_sql
__END__
output:
SELECT "companies".*
  FROM "companies"
 WHERE ("companies"."transaction_from" <= '2022-08-24 08:07:43.822990'
   AND "companies"."transaction_to"    >  '2022-08-24 08:07:43.822990'
   AND "companies"."valid_from"        <= '2022-08-24 08:07:43.822990'
   AND "companies"."valid_to"          >  '2022-08-24 08:07:43.822990'
    OR "companies"."transaction_from"  <= '2022-08-24 08:07:43.819944'
   AND "companies"."transaction_to"    >  '2022-08-24 08:07:43.819944'
   AND "companies"."valid_from"        <= '2022-08-24 08:07:43.819944'
   AND "companies"."valid_to"          >  '2022-08-24 08:07:43.819944')
```

The bug did not reproduce when `Timecop.freeze` was used because the structure of the `Arel node` is different due to this difference.
( When using `Timecop.freeze`, `Arel::Nodes::And` was not included )

Reply:
Ah, understood. Thanks.